app/services/course/assessment/answer/auto_grading_service.rb

```diff
@@ -0,0 +1,37 @@
+class Course::Assessment::Answer::AutoGradingService
+  class << self
+    # Picks the grader for the given grading, then grades into the given
+    # +Course::Assessment::Answer::AutoGrading+ object.
+    #
+    # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store grading
```

Comment:
I think it's not a good idea to use input as well as output
And how about change the params to `answer` ? I think `grade(answer)` makes more sense. 

My two cents, what do you think ? @lowjoel